mm2src/mm2_main/src/rpc/lp_commands/lr_swap.rs

```diff
@@ -0,0 +1,263 @@
+//? RPC implementations for swaps with liquidity routing (LR) of EVM tokens
+
+use super::one_inch::types::ClassicSwapDetails;
+use crate::rpc::lp_commands::one_inch::errors::ApiIntegrationRpcError;
+use crate::rpc::lp_commands::one_inch::rpcs::get_coin_for_one_inch;
+use coins::lp_coinfind_or_err;
+use lr_impl::find_best_fill_ask_with_lr;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::{map_mm_error::MapMmError,
+                     mm_error::{MmError, MmResult}};
+use types::{LrBestQuoteRequest, LrBestQuoteResponse, LrFillOrderRequest, LrFillOrderResponse, LrQuotesForTokensRequest};
+
+mod lr_impl;
+mod types;
+
+/// Find the best swap with liquidity routing of EVM tokens, to select from multiple orders.
+/// For the provided list of orderbook entries this RPC will find out the most price-effective swap with LR.
+///
+/// More info:
+/// User is interested in buying some coin. There are orders available with the desired coin but User does not have tokens to fill those orders.
+/// User may fill the order with User my_token running an preliminary LR swap combined with the ordinary dex-swap (using 1inch as the LR provider).
+/// Or, User may convert the tokens from the order into my_token with a subsequent LR swap.
+///
+/// User calls this RPC with an ask/bid order list, base or rel coin, amount to buy or sell and User token name to fill the order with.
+/// The RPC runs 1inch swap quotes to convert User's my_token into orders' tokens (or backwards)
+/// and returns the most price-effective swap path, taking into account order and LR prices.
+/// TODO: should also returns total fees.
+///
+/// TODO: currently supported only ask orders with rel=token_x, with routing User's my_token into token_x before the dex-swap.
+/// The RPC should also support:
+/// bid orders with rel=token_x, with routing token_x into my_token after the dex-swap
+/// ask orders with base=token_x, with routing token_x into my_token after the dex-swap
+/// bid orders with base=token_x, with routing User's my_token into token_x before the dex-swap
+pub async fn lr_best_quote_rpc(
```

Comment:
Is there a way to get streamed events for the best quote in the future? I see that only fusion+ has a [websocket API](https://portal.1inch.dev/documentation/apis/swap/fusion-plus/fusion-plus-sdk/for-resolvers/web-socket-api) so it's probably not possible. But we can stream it periodically I guess instead of GUI polling it. Please add this to the issue checklist.

Reply:
A somewhat related comment. Price changes are very fast, so we would have to stream it every second at most, we can make it configurable, but this is where slippage comes in when actually filling the order as an error can be returned if execution price is different from the one shown to the user (the price minus slippage of course).

Reply:
The OHLC API returns token prices with granularity of 5 minutes minimum.
Getting a quote each second ver 1inch API I guess would cost much their infra fees (unless we setup our own service which can interact with their contract directly, with the needed interval)